src/main.cpp

```diff
@@ -3,59 +3,59 @@
 #include <DataCollector.h>
 #include <DigitalSensor.h>
 #include <AnalogSensor.h>
-#include <switchSensor.h>
 #include <Sensor.h>
 
+constexpr uint32_t THROTTLE_1_ID = 3;
+constexpr bool THROTTLE_1_CRITICALITY = true;
+constexpr uint8_t THROTTLE_1_PIN = 39;
 
-FlexCAN_T4<CAN2, RX_SIZE_256, TX_SIZE_16> can2;
+constexpr uint32_t THROTTLE_2_ID = 4;
+constexpr bool THROTTLE_2_CRITICALITY = true;
+constexpr uint8_t THROTTLE_2_PIN = 40;
 
-constexpr int THROTTLE_1_ID = 3;
-constexpr int THROTTLE_1_WAIT = 10;
-constexpr int THROTTLE_1_PIN = 39;
+constexpr uint32_t THROTTLE_INTERVAL = 10;
 
-constexpr int THROTTLE_2_ID = 4;
-constexpr int THROTTLE_2_WAIT = 10;
-constexpr int THROTTLE_2_PIN = 40;
+AnalogSensor throttle1 = AnalogSensor(THROTTLE_1_ID, THROTTLE_1_CRITICALITY, THROTTLE_1_PIN, THROTTLE_INTERVAL);
+AnalogSensor throttle2 = AnalogSensor(THROTTLE_2_ID, THROTTLE_1_CRITICALITY, THROTTLE_2_PIN, THROTTLE_INTERVAL);
 
-constexpr int BRAKE_ID = 2;
-constexpr int BRAKE_WAIT = 10;
-constexpr int BRAKE_PIN = 21;
+constexpr uint32_t BRAKE_ID = 2;
+constexpr bool BRAKE_CRITICALITY = true;
+constexpr uint8_t BRAKE_PIN = 21;
+constexpr uint32_t BRAKE_INTERVAL = 10;
 
-constexpr int SWITCH_ID = 1;
-constexpr int SWITCH_WAIT = 10;
-constexpr int SWITCH_PIN = 38;
+AnalogSensor brake = AnalogSensor(BRAKE_ID, BRAKE_CRITICALITY, BRAKE_PIN, BRAKE_INTERVAL);
 
+constexpr uint32_t SWITCH_ID = 1;
+constexpr bool SWITCH_CRITICALITY = true;
+constexpr uint8_t SWITCH_PIN = 38;
+constexpr uint32_t SWITCH_INTERVAL = 10;
 
-AnalogSensor throttle1 = AnalogSensor(THROTTLE_1_ID, THROTTLE_1_WAIT, THROTTLE_1_PIN, true);
-AnalogSensor throttle2 = AnalogSensor(THROTTLE_2_ID, THROTTLE_2_WAIT, THROTTLE_2_PIN, true);
-AnalogSensor brake = AnalogSensor(BRAKE_ID, BRAKE_WAIT, BRAKE_PIN, true);
+DigitalSensor startSwitch = DigitalSensor(SWITCH_ID, SWITCH_CRITICALITY, SWITCH_PIN, SWITCH_INTERVAL);
```

Comment:
This is an odd case we haven't talked about o think. The switch sensor class was a kind of band aid kinda not fix for a hardware issue with the switch. So for the time being it being a switch sensor and not a digital sensor is critical. (Unless digital sensor was tweaked but I can't remember)

Reply:
Last we spoke you said you didn't want to have a separate switch class. I can see that you are doing some funky debounce math that probably would warrant it's own sensor implementation. Do we even think this will be necessary though? I though you mentioned it was likely a hardware thing that would be fixed when we get a new swtich.